Quran/SQLiteSearchableAyahTextPersistence.swift

```diff
@@ -119,6 +160,29 @@ class SQLiteSearchableAyahTextPersistence: ReadonlySQLitePersistence, Searchable
         }
         return cleanedTerm.lowercased()
     }
+
+    private func generateRegexPattern(term: String) -> String {
```

Comment:
this generates the regex we run on each result from the database - it's basically the search term, with any of the possible characters that have alternatives (the keys of the `replacements` table) replaced with an optional set of any of their replacements.

Reply:
What happens if the user includes any of the reserved regex characters? I think the search term is cleaned up before getting to this point, right?